bitcoin/src/bip32.rs

```diff
@@ -155,9 +211,9 @@ impl ChildNumber {
     /// [0, 2^31 - 1].
     ///
     /// [`Hardened`]: #variant.Hardened
-    pub fn from_hardened_idx(index: u32) -> Result<Self, Error> {
+    pub fn from_hardened_child_number(index: u32) -> Result<Self, Error> {
```

Comment:
I think there might be a difference between index and child number. It looks to me that child number contains the hardened flag while index doesn't. So index was probably correct here. We're also storing index in the `*ChildNumber` which is confusing. (Also maybe if integer validity ranges stabilize in the future we can get better optimization changing it.) I don't care to change the layout in this PR though.

Reply:
Cc @apoelstra, shall we revert back to `from_*_idx`?

Reply:
Oh, yeah, I guess `index` is correct then. That is confusing.

Let's expand `idx` to `index` though.